src/util.h

```diff
@@ -53,6 +53,13 @@ namespace node {
 
 #define UNREACHABLE() ABORT()
 
+#define ASSIGN_OR_RETURN_UNWRAP(TYPE, PTR, OBJ, ...)                          \
+  do {                                                                        \
+    PTR = Unwrap<TYPE>(OBJ);                                                  \
```

Comment:
Mind giving me an assist here? I've changed the macro to
```cpp
#define ASSIGN_OR_RETURN_UNWRAP(ptr, obj, ...)                                \
  do {                                                                        \
    ptr = Unwrap<decltype(*ptr)>(obj);                                        \
    if (ptr == nullptr)                                                       \
      return __VA_ARGS__;                                                     \
  } while (0)
```
But the compiler gives me:
```
../src/js_stream.cc:186:15: error: no matching function for call to 'Unwrap'
  do { wrap = Unwrap<decltype(*wrap)>(args.Holder()); if (wrap == nullptr) return ; } while (0);
              ^~~~~~~~~~~~~~~~~~~~~~~
../src/util-inl.h:194:11: note: candidate template ignored: substitution failure [with TypeName = node::JSStream &]: 'Unwrap' declared as a pointer to a reference of type 'node::JSStream &'
TypeName* Unwrap(v8::Local<v8::Object> object) {
~~~~~~~~  ^
```
As soon as I get this figured out I'll implement your other suggestion.